src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.macios.cs

```diff
@@ -157,12 +157,19 @@ static bool TryGetCurrentPage([NotNullWhen(true)] out Page? currentPage)
 		}
 
 		// If not using Shell or a Modal Page, return the visible page in the (non-modal) NavigationStack
-		if (window.Navigation.NavigationStack[^1] is Page page)
+		if (window.Navigation.NavigationStack.Count > 0 && window.Navigation.NavigationStack[^1] is Page page)
 		{
 			currentPage = page;
 			return true;
 		}
 
+		if (Application.Current != null && Application.Current.MainPage != null && Application.Current.MainPage is Page mainPage)
```

Comment:
FYI - Once  https://github.com/CommunityToolkit/Maui/pull/2215 is merged, we will need to modify this to use `Application.Current.Window` instead of `Application.Current.MainPage` because `MainPage` is deprecated in .NET 9

Reply:
Also FYI  - you can ignore the current build errors in our CI pipeline. Those too will be resolved when we merge https://github.com/CommunityToolkit/Maui/pull/2215

Reply:
Sounds good.  Is there anything I need to do for now?

Reply:
We are in a code freeze until we merge #2215. 

You'll need to fix this before we can merge this PR. It will not be merged before #2215.